vumi/persist/riak_manager.py

```diff
@@ -122,7 +122,17 @@ def from_config(cls, config):
             mapreduce_timeout=mapreduce_timeout, store_versions=store_versions)
 
     def close_manager(self):
-        self.client.close()
+        if self._parent is None:
+            # Only top-level managers may close the client.
+            self.client.close()
```

Comment:
Should we raise an exception if we don't close the client because it's not top level? Seems a bit strange to call `close_manager` and for it not to close anything, or maybe that's what should happen. Just a thought.